tap_core/src/error.rs

```diff
@@ -39,6 +39,16 @@ pub enum Error {
     AdapterError { source_error_message: String },
     #[error("Failed to produce rav request, no valid receipts")]
     NoValidReceiptsForRAVRequest,
+    #[error("Previous RAV allocation id ({prev_id}) doesn't match the allocation id from the new receipt ({new_id}).")]
+    RavAllocationIdMismatch { prev_id: String, new_id: String },
+    #[error("All receipts should have the same allocation id, but they don't")]
+    RavAllocationIdNotUniform,
+    #[error("Duplicate receipt signature: {0}")]
+    DuplicateReceiptSignature(String),
+    #[error(
+        "Receipt timestamp ({receipt_ts}) is less or equal than previous rav timestamp ({rav_ts})"
```

Comment:
`less or equal` does not correspond to the code (`strictly less`), BUT, turns out the code is wrong! So perhaps let's fix that line of code in another PR and merge it, then update this branch?

Reply:
#115

Reply:
#115 is now fixed on the main branch. Could you update to latest main?

Reply:
done !